src/modules/CreateDrops/contexts/DropFlowContext.tsx

```diff
@@ -0,0 +1,58 @@
+import React, { createContext, PropsWithChildren, useContext, useState } from 'react';
+
+import { IBreadcrumbItem } from '@/common/components/Breadcrumbs';
+
+import { IFlowPage } from '../types/types';
+
+interface DropFlowProviderProps {
+  flowPages: IFlowPage[];
+  breadcrumbs: IBreadcrumbItem[];
+}
+
+const DropFlowContext = createContext({
+  currentPageIndex: 0,
+  flowPages: undefined,
+  breadcrumbs: [],
+  onNext: undefined,
+  onPrevious: undefined,
+  currentFlowPage: undefined,
+});
+
+/**
+ *
+ * Context for controlling navigation of the drop flow pages
+ * and sets layout information such as breadcrumbs, pages
+ */
+export const DropFlowProvider = ({
+  children,
+  breadcrumbs,
+  flowPages,
+}: PropsWithChildren<DropFlowProviderProps>) => {
+  const [currentPageIndex, setCurrentPageIndex] = useState(0);
+  const [maxPages] = useState(flowPages.length);
```

Comment:
Do you need this as a state? I think you can simply do `const maxPages = flowPages.length`. Once the prop `flowPages` change, the component will re-render which will update the maxPages const so it should work the same way